using System;
using System.Collections.Generic;
using System.Linq;
using System.Text.RegularExpressions;

namespace AdventOfCode2017
{
    public class Day9Part2 : ISolution
    {
        public int Solve()
        {
            bool inGarbage = false;
            int groupNesting = 0;
            bool cancelNext = false;
            int score = 0;
            int garbageCharacters = 0;

            foreach (var chr in Input)
            {
                if (cancelNext)
                {
                    cancelNext = false;
                }
                else if (inGarbage && chr == '>')
                {
                    inGarbage = false;
                }
                else if (!inGarbage && chr == '<')
                {
                    inGarbage = true;
                }
                else if (inGarbage && chr == '!')
                {
                    cancelNext = true;
                }
                else if (inGarbage)
                {
                    // Nothing else has meaning in garbage, just count the garbage
                    garbageCharacters++;
                }
                else if (chr == '{')
                {
                    // Entering a nested group
                    groupNesting++;
                }
                else if (chr == '}')
                {
                    // Exiting a group. The score of this group is how deeply it is nested
                    score += groupNesting;
                    groupNesting--;
                }
                else if (chr == ',')
                {
                    // I reckon commas are basically red herring. Let's hope so anyway...
                }
                else
                {
                    throw new InvalidOperationException($"I don't think we're supposed to see a {chr} here...");
                }
            }
            
            return garbageCharacters;

        }
        
        private static string TestInput = @"<{oxi!a,<{i<a>";
        private static string Input = @"{{{{{{<!>!>,<o!>},<a,xi!!!>i!!,!>,<<e<i<<>,{{{<!>},<!><!>,<!!!>!!!>!!!>{x!>},<!!!<!>},<oi!>!,'>}}}},{{<{!!!>},<!!,!!!>!!!>!!e!a!!!!<!>},<!!!>u!>,<!>!!u!!!!!>},<!>!>,<{>}},{<xa!!!!'!>},<!}x!>!!<o}i!<>,<<e,<eo!a}!!x!>,<!>!>!>,<e{xex>}},{{<!oi'<!o!>ue>,{<!>},<i!>,<i>,{<'>}}}},{{<xa,!>},<!!e!>,<e!!}!!o!!!>,<',}>,{{<i!!!!o!!!>!}!!!!!>!a!!!>!>!!a!!!!,!!u!!!>,<>}}}},{{{<{i!!<!!!!e!>>}},{{<!!!>},<ei>,{}}}}}},{{{<eaieia,!{iox{!!}eu!{{!!e'>,{<>}}},{{{{<o>},<e!!o,!>},<xu{!>,<!>},<a{}{!u>}},{{{<!!u'ioi''!>},<xu>},<'oa!!<,'!!x!!!!!>!!!!!>{!!!!aa>},{{<u}x,!!i!!!>!>!!!>!>!!!>'}!o{!!{>}},{<oiau>,<!!!>!><!>!!<!!!>!!<i!a}!>},<<}a}>}},{<eeo!!}!!!>e!>,<!!!>!!<{!!<i{!>i!>,<>}},{{{},{<!!!!!>},<x!!!>!>},<o{!>,<,ea>}},{{<},xaxoeie!>},<ie'!!oa!!<>}},{}}},{{{}},{{<}e<i!!!!io!>},<x!!ex<x>},{{<'!!{!>},<ax>}},{<u!!i!!o{>}},{{<xoe!!!>xo{>},{{},<!>,<!!,!}!>!!!!!>>},{}},{{{{<!!{,,}!>!>,<!>},<!{a!<e'!>,<>}},{}},{{{<!!ox'!!!>xi{''>},{{},{{<!!!>},<x}<x,!>>,{{},{{}}}}}}}},{{<u}x!!!!!!o!!x!>,<!>,<eiu!!!>,<o>},{<!!<!!e>,<e,<<<u!'>}}}}},{{{{{{}},{}},{{{<!!!>x!!!!oa!>},<{a<}>,<'ixa!!!>>},{<x!xea!}e!!!>!>},<}o{>,{<e!>,<!!}xi!!,!>ie!{{!>},<!>},<'!>},<e!!!>!!!>x>}}},{{},{}},{{{<i!>},<}!>},<{!ae!>,<!>,<!>,<!!!>'i!!>},{}},{<{!!!>!!!>a,!!!}!!!!}!>,}e}<{!!'!!!>!>},<>,<!>,<!>xxe<!!u<!!,e>}}}},{{{{{},{<!!o}!>,!>!!!>>},{{{{<,>}},<x,x!>},<!!!!!!!>!>!i!>,<{!!!>}!>,<!!!>{ou>}}},{{{<!!e<!'>},<!a!>!u!!!>!>,<!!!!!!!>{x<<!><!!!!u!!!!u>},{<!!!i!!!!!>uo!!!>,<a<>}},{},{{{{<!x{}!!!>{a{i{a!>,<!!!>'>},<x{!!!>}!a!!!<!>,<>}},{{{<>},{<!!!>!,{{>,<{!!!>u!oi!x{''oo!!!o,i}ao,e>},{{<<!!<!>},<!!!,>},<o!>,<}!!!>!e}o!>},<<>}},{}},{{{},<!!!>{!>},<!!!>i!ex!!!>!!x,!>},<uuo!>},<'!!>},{{<>}}}}},{{{<{x!>x!!e!x}<a!>>,{}},{<ui!>},<e!>!!!}u!<!>,<,{!!o!!!>!>,<!,>},{{{<a!!a!{!!!!ux!!a,!!o!>,<}ux>}},{{}},{<!>},<!!!<}a!>,<u!!!>x,!!!>xx!!ee,!>},<!>},<u>,<,'<u!!!>!!<!!!!x!!!!!>aa!!>}}},{{{}},{{<a!>!>!>,<>},{<}a{i<!!!>},<'!>,<xe{!>,!!u,!!!!!>>,<!a,!>!}!o>},{{{<u!!uox!>,<ax,!!!>x>},{<,eoe>}}}},{{<!!!!!>!>},<ue!!!>,,!>u!!!>e'<!!!>!>,<ae!>>,<!!i}!>,<<'i'!!!>a!!u!!!>>},{<,ui{>}},{{{{}},{<{!!!>,<a{}!>ox>,<!!!!!uuxa}'io!>},<!>x}e!>},<>}},{},{}}},{{},{<>,{<'{u!>!!!>},<{!!!!!>ix!!>}},{<!!i!!''!>!!!>!!!!!>!!u>}},{}},{{{<,!>!!!>!>,<a{a!!}!!u,'!>,<}!>},<<!!>,{{{<i!>},<,!!!>>}},<>}},{<<xa!!x!>,<!iuao!>!!!>u>,{<i'!!!!<,i!oiea!!!>>}},{{}}},{{{{<,!>,<!>,<ua,u!!{{!>,<!!,ui,!>},<io>},<!o!!!!!!!!!!!!xuuux<{!!!>}>},{{<!!!!!>i>},{<ex<'!!xxe!>,<!x>}}},{{{{},{<!>>,<a!>},<x!!!>!>e{e<x<!>,<eu!!!>,!!!>!!i>}},{<}}!!!>!>!!}e!>,<!!!>,<x}!!e}{>,{<}u>,<u!eeo!>,<au!!!>e,>}}},{{{{},{{{<e,!>}!>o!,<!>},<}!>,<!>e!!aa<x,!!'>}},<{'i!>!!!!u{}!!!>},<u!!!!}>}},{{<i!!!>,<'!{x!<a!!'!!u!!o!o<!!!>{!!!!ei!>>}}},{{{<aea!!!!{!>!!i!!!!!!!a<}xi}!>},<iu,a>}},{{<ux!!<!>a<>}},{<!!!!u{}!>},<!u}u<!>,<oe{!!!>!u}u>,{<x!!iu!!!>}>}}},{{<{i<<!{x!i{x!>,<!!ou>,<oo!>{>},{},{{<e!!{!iaui!!e!!!>>},{<!>,<!>i!!{'u<xxu!>ox{}!i!>}!!,>}}}},{{<{u!!!!!>x{!>!!!!!>!!!i!>aue{,<,>}},{{{{}}},{<!>,ex},i!>},<!!oe!!xa<>,{<x!!}!>,<u!!!>o!!!!x!!u<!>{!>,<!<i!!x!!x>}},{{{{<i<!x!!x!!!><i<!><x!u!>},<a<!i!!!>,<!a!!!>>},{<i{!!!>a<u>}},<!>}o}!>a!!},!!!>!>}!!!>,<>},{<{!>,<i'u!!o}!auiou,!!}!>>}}}},{{{<<!!!>!!!>,<!!!>},<!>,<i>}},{<!>,<!>,<!!!!eu}!>},<!!}!!}>,{<{!!!!!>},<!!!!!!!!!>{!>,<},x>}},{<uu!!}e{!i!>},<>,<}!!o!>!!!>a!!!>!{<'a!>,e!!!>u>}},{{},{{{<>},{<!!!!xo!>e!!>}}},{{{<!!!>a!!!>'!!a!!{u!!!a>}},<!>},<!>,<u!!,'x!>!!!>},<>}}},{{{<!!>}}},{{<!!!!!>o!!!>o!!!!eu{x}!o,e}>}}},{{{{<<!!!!{{!>},<!>,<ex!!!!!>>}},{{<!}!>},<!!!>,<!!!>>},{{<!!i<>},{{<,x!>},<>}}}},{{{<!}a!i'!!'i!>>,{{<eii!>},<!>},<!>},<,u<<!>,<ei!u!!!>},<!!a>},<,{!!!!!>,<!!'>}},<!>>},{{{<,!>!>ou!!u!>,<i}'u!!!!ix!>,<i!>,<{!!>},<!!ux}}'i!>},<o'!}!!!>e'u!>io!,>},{<!>!!}!>},<!!!>!>},<!>!>,<'!>,<i!>>}},{{}}},{{{<!!!>,<}'{!i}!>x!!uxa,a!!!><>}}}},{{{<u!!}xxe!ee,{!!,>,<,,!>},<>},{},{{<uo!>>},<!>,<eo'!>},<a!>,<!<ou!!!>,<,<>}},{{{{{<!<!!!>!!!>x!xa!>,<!!,!!!>!>},<u{!!i!>},<!eu>}}}}},{{{{},{{<!>,<!>},<{!!!>i!>},<>},{}}},<,uu>},{}}},{{{},{}},{{{<xa!>,<<{!>!>},<'!>!>,<<!>,<}'!>!>,<>},{{<o,a!!!>}!!!>>}},{{<eoi!>!>!!!>},<!!!>!!!>}}!ee!!!>!!!>,<!x!!u!!!>>}}},{{{{<!>,<x!!{'}!!!!!>'!xx!!!>!>u!!o}!>},<>}}},{<!>,<>,{}}},{{<o'<>,{<!>,<!!!>!!}i,!!ai!>},<oo!}!>},<!!>}},{<{!>,<a>,<u!!>}},{{{{}}},{{{}},{<'!>ai,!>},<!!!!!{!>oei!!!>x!>,<'!>},<o>,{<!>,<i,!!<!>,<!>'!!!!!!!>!!e,!!}o<>}}}}},{{{{<!!!!!!{!>,<!!!>,<}!>,<a!>}!>},<!!!>i!>,<'i>}},{<!!<!>{e'i!!o!!<u>}},{{{<!!a!!'ea!>},<}{a!!!>,<e!>,<!!!>'!!!>ux>},<!>e!>},<u!>,<!i!>!!<x!!!>,<{u!>,<!!xx,u>},{<x}',!ui,!!!!!>>},{}},{{{<e!>,<'!!!>!>,<!!!>},<,eu,>}},{<{x!>'x!>,<}!>,<,!!!>}o!!a!>ei'o>,{{}}},{{<!><<}!x,<i!'<!!>},{{},{<!>!>},<!!!>},<>}},{{<<u}!>},<<!x,'xx>},{<{!!!!,!>!>},<o!!}a,!>},<'o<o!>},<!>,<>}}}}},{{{},<,!,!>},<!xo!!!>>},{},{{<!>,<!'!!'i!!,o'!!!>,x>}}}},{{{{<<,!!!!!!!!!>},<!!!>!!<!xa!!!>,'!!!!a!>,<>}},{<!>},<!>,<!!}!!{xu<{o!>},<!!o,>}},{{},{{{<!>,{!!!'>},{{<x!>},<!'!{!u>},{<!}o!!!>eai!u}!>},<!>},<<,ae>}}},{<!>,<<!>,<>}},{{<}!'!!o!!!>a'!!!>!>,<x!!!!!!x!!!>},<!!!!!!!>>,{}},{<!>!><{>}}},{{{<}!!,x>}}}},{{{},{{{{{{<xi!>},<',!>x'!>x{!!x!>e!>u>},{<'}>}},{{<,i{>},{<{!aa}u!!!>!!!!ia'!>,<!!x>}}},{<x!x>}},{},{{<!!!><{o!,'}a!!u!>},<}x{<>},{<a!{i!>!}!!'ea!>,<!!!!!>!!!><i<!!!>},<!>},<>}}},{{{},{{<!>},<!{e!!!!}xa!!o!!<o,!!!>!>},<>},<>}},{{}},{{<!!,a,}!!a!!!!!!'!xa!!!!!xau>,{<<!!!>!!!!!><!>,<a!!u!!!>!!e!!!!',{{o!>>}}}},{{{{<!!!!!>,!!!>,<!>},<!!<!!!!!!,!!>}},{{{<<!!!>xu<!!!!,!>,<!!,>,<!!a!>,<>},{<!>uou!>>}},{<x,!!!>,!>>,<a'ia!>},<!,{!!!i!>},<>},{<},!>},<!!!>}!>},<!!!>!>,<o!!ou}>}}},{<>,<''!>,<oix!!!>},<<i{{!{e>}}},{{{}}}},{{{{<a,i!>,<a!>},<>,{{<!>!!!>}!oe!<uei'},!!!>},<!!!>u>},{{<!{!>e<>}}}},{},{{{},<!!,}!!!!!>,<i!>u}e}!!ux>},{{{<}o'!>',<!>!>},<{,''>},{{<oe}{u<!>!>>}}},{<!!!>!!!iaiu<!!x!>,<'!>},<u!!ax>}}}},{},{{<{!,i!!!>,xi!!<x!!i<!!!>!>,<>},{{},<{!>,<!!!>u!!!!!>!!}uo!!!!a!>,<{x>},{<>,{<<xxoa!>},<}ao{i!a!!!>x>}}}},{{{},{<!!<oa!!x}!>,<!>!!i!!!!!>,<!>},<!!e!>},<o>}},{{{},<!!!!!>!!u>}}},{{<!>!>},<!!!>x>,{{<!o!'x!!!!'u!!!>,<xo}>},<x>}},{{<i<!>},<!'a!!!>!>!>>},<!x},}i}!!!>>},{{{<i!>}a<>}}}}},{{{{}}},{{<'!>,<!!x!>,<!!}!!'!!eui!>},<}>},{{{{<x!>o'e,eu!!i!!!!{!!o!>>}},{<>}},{<{!<!!{u!>},<>}}},{{{{<>},<!!!>uee!<!!>},{{<!!!>iu}!>>}}},{}}},{{{}},{{{{},{},{<o<!>'ea}>,<!>o,a'!>,<uxi{e}o{>}},{<''{<!!}!!{i<oa>,<!!,xiio{}>}},{<!>,<{oi!>},<e!!!>uaa!>},<!!!>u!!>,<a>},{}}},{{{{{}}},{<<!>},<,e!>,<{<a>}},{{{{}},{{},<!>!!!>!>!!!>{!>,<!!!>,<a,{{!!!>,<'!>!!{>}}},{{},{{}}},{{{<x!a!>,<!>},<,io!!!>!>,<u'!>,<!!!>!!!>},<!}!!!>>},{{<!>,<!>,<'u!!!>,u!i!!x!!!>x!!x>}}},{<!!!>!!!>!>},<<>}}}}},{{{{{<'au!>,<!!!>},<!>>}},{<u{!e!,u!!a!e!>,<{ii!>},<!!!>!!>,<!}!!!>{'aou{!>},<ox>}},{{{<!!!>>},<x!>>}},{{<uo!>!!!e>,<!>!!}!>},<o>},{{},{<o>,{{<!!!>},<<au>},{}}}}},{{{<x!>!!u!>}i!!!>ua!>},<!!o,e!>},<a,u!>>},<,o!!!>},<!>,<!!i>},{<!>,<!''a!!<<o,!>},<{'!>>,{}}}},{{{{<o!>a!>,<{>}},{}},{{<o}'!>ui!!}'e>,{}},{<x!!i!!,a!!!!ui,>,{<,!>},<e>}},{{<oa!>!>,<xxo,xoo>}}},{{{{},{}},{},{<!>{!>,<!!!>!!!e!>{e!!>,{<{!!!i{x!>},<uu!>,<>}}},{{<!!{x!>,<,!!'>}},{<!>},<xox!>,<iox!!!!,'>}}},{{<e!!!!{oia}'''}!!!>,<>,{<'>}},{{},<!!,!>},<<e!!!!!!!!!>!>,<ua!<a!!i{,!!{'o<>}}}},{{{<<<!!<!>!>,<}x>},{{{{<!>'{>}}}},{{<,e<>},{<i>}}},{{{{<a}}!>},<o,!>!>},<{!>u}!!!>{a<,!>'>,<o<<i>}}},{{{<o}x!!,>,<!!!!!!<o>},<!ua!<<!>,<<!>},<!!oa{!>o!>,<!!!!!>xx>},{{{}},<!u}a!!!!>}},{{{<>}}}},{{{{<!o}a!!!>x!e!>x>},<!{!!!>!>!>,<!!{xie!u!!{>},{{{<!,!'!>,<!!!>},<x!>},<>,{}},{{<>},<o!!a}!}a!>},<a!!!>!!i!>,<!>},<!>,<!>i>}},{{<{i<x!!!>{!!aa!!!>!!!>ai!>'ui!{!!!>>,{<<!!!>,<}!!!>!>!!<!!!>{!!!>!iiu>}},{{<i!>},<a<x!>,<aao}<!!!o!>'e>,<!!{x,!>,!>,<!>},<!>},<ioxu!!!>!>,<a{!>},<'>},{{<!'!!oa!>,<!!!><!!!>!!!!!>},<{<'!o},>},{<!u!>,<{<!>},<{}<!!ee!>,<ux>}},{<uu!o!>,<!,!>,<!u<>,{<'!>},<i'!>,<ux!!!>!!xx'!!a!>!>ui'!>>}}}},{<{!>,<e<!!!>,<i!>>,{}}}},{<'!!!>,<>,<}ix!>,<!!!!!>!!}!!x!>},<!>!a!!iux>},{}}},{{{{{{{<{eee!!!>}!>,<!!,!>'>}},{<!>},<u>}},{{<!!{!!{!e{}'!!<!!ui!>}!!'!!,o!>},<}o>,<!x!>},<!!!>},<!!!>!{<!>},<!!!>,!u'u}o<>}}},{{<x'}',!a!!!<!!!!!><!!!!!>,<!>},<!!}!>!!!!u{!au>,{{},{<,,!!!>,<x!!>}}}},{{<ioe!>!!!>e!i,,>}}},{{{{{{<!!!>},<!!!>e!oa{}!>},<x}!,,<!>,<e>}},{{<o>,{<!><<!!!>x>,{<!>,<!>,<'e{e!!!>!!!>},iu!!!>x!i>}}},{{<,!!!>},<uo!!!>oax!a!oa!!!>>,<!!<i!u!>},<ix!>,<xe!>,<<!!,{a>}}},{{{<!!!>!>,<!>,<{!!!!!>i!>},<<x!>!!!!>}}}},{},{{{{},{{{<!>'{'!!!>},<!!>},<!!}>},<!!''{{>},{{{{<<!<iue},u}!!!>!>},<!!!>''>}},<i!!!!!!x!>!>,!!!!!>},<,x}!e!!!!!>>},{{}}}}},{{{{{{}},<xi}<!!!!o,x}!>>},{{<'a!>},<!!e!>},<>,{<e!>,<!>i,!>},<'<!>},<!>>}},<,!>,<o!>,<>}}},{{},{}}},{}},{{{}},{{},<{<o!>!>ix'!>!!!!<!>,<!<{!ex>}}},{{<x!!>,{<!>},<u'!>}{!>},<}!!{!>,<!,>}},{<!o<a<}!!!>>,<!!!>i!o!!!!}x!!!!u{!>},<'!>!<!>},<>},{{<!!o}!>},<o!{}!!i!!!>!>,<o!>,<!>},<{{a>}}},{{{<u!!o!!!>!!!a!i!>,<!>},<<!!!!<}!!ua!>>}}}},{{},{{},{<ix,}>}}},{{{{<,!!!>!!!!x}!>},<!>,<!!{!>},<}>},<!>},<!>oeo,!!{o<e>},{{},{<!{!>},<!!!!!>i,>}},{{{{<a!>,<o{iix!!i',o!>,<>}},{}},{{{}},<!!!>!>,<!>,<{>}}},{{{},{{{{}},{<!!!'!<!>},<>}},{}}}},{<<!u!!!!!!!!!>!>!!!>!!eiu>,<!>,<<!>},<,!!!>'{'!!xao}u!!!>'o>}}},{{<ux!>},<!!aae<!!!>!!!>},<'>,{<!!o,!>a!!!>,>}}}},{{{{<!>,<x>},{{{},{<>,<o,!xiei!>!>},<}!!!>!!!!!!!>!>,<>}}},{<!!!>!!!eao!>!!!>i<{o!>,<!!!>},<e>}},{{},{{{}},{{{},{<!!!>!!!>!o!!eu!!<<e<>}},{{{<x!!!>!>x!e!>ex!!eu!>},<e!>},<'>},<!>},<<oa!>{!>},<!!!!ixi}}!!'<!!!>},<>}},{{{}}}},{{},{{{{<xiu!!!aoo,!x>,<o!!!>!!<!>!>,<oi!ao!!!>!>,<!!!>,,>},{{<}!>},<!>!!!>},<i'!!!,x>}},{{<!!!>a!!{,!!!x!>,<xi<iui!!!>!!!a}!''>}}},{},{{{{},{<iii!>!<,<!e!>}}xo!!!!!>},<<}!!!>',>}},{<o{!>,<!>!u!>{!}!!e>}},{{<!>eo>}}}}},{{},{<e!>,<a!!!e{a!!e!>!i,'!!!>exaa!>,<>}}}},{{{},{{}}},{},{{}}},{}},{{{<!!!>!,u}!>uo!>},<,x<!>},<a,!>},<>},{{}}},{{{},<<!>},<a{{!>,<e!!<!}{!!{!!!>},<'<>}},{{{},{{<a{o!!!!!>',,!x!>eia!>},<!>,<{<,>},<e!>,<>},{{<!,>},{<a!!!>,<,'i!!!!,!!!>i!<!!!!<,!>},<,!>},<>}}},{{<!!i!>!!}o!>},<!!!>!!!>e,{!>},<ae!!'>,{{<!>},<>},<!!!>eea!!{!>},<!><!!!>!!!>,<>}},{<i{!!!>!>},<o,,iu!!!>!>,uee'>},{{<o!!<e!>i!!e!!!>>},<exio!>},<!a!>,<{!>},<!>e!>eu!!!!!!!!!>,<o!>},<'!!>}},{{{<!!!>'ooeaa!>,<{>}}}}},{{},{<!<{uo!!!!!{!>!uo!!o<uex!!ia,'>}}},{{{{<>}}},{{{{}},{{{<!x!!<!!!>x>},{}},{{<a!!}uxi!!!>},<<>}},{{}}}},{{{{{{{{<!!ix!>},<u!!!a{i!!x>}}},{<!!!>o!!o<!<ui!>!!{o!!u{>}}},{{},{<,!!ueua,{o{!!!>,<<>}},{}},{{<>},<}'!>,<!!ui{o{!!!,!!,ue!!eu!!!>>}}}}},{{},{{{<eo!e!>!!!>!>!>,<i!!a!!!>},<!<!!e!!u,!>,<o,>,{<'!!!>,<<a!!!>,<!>},<o!!{}!!!!'!>,<o}<>}}},{{{<}!!,!!!!!>!>},<u{a!!{o,o>}},{{<>},{<!!i{!!!>!>u>}},{{},{{<,x!!!>!!{>,<i!>},<<a>},{<!!!>u}'!!e!x{''o>}},{<!u<oa!!<>,<uu>}}},{{{<}>,{<e!!!>!>},<!>,<!!!{}!!e!>>}},{<u,>},{<a!}x!,!,!!!,uuiiu>,<!>,!>},<u{}<x!>,<'!!'!>},<ex}ui'>}},{{{{{{{<ua!>{iu!!',!>},<!e'e!>,<>}},{{<o!>,>}}}},{<!>!>,<!!!!}e>}}},{}},{{{<<!!<!>},<}{!!auo!!!!!><e!!e'<!!!<>},<!><a'}',eaei!>>},{{},{},{{{<!!euui!!!!!>},<'{oi>}},<o!!!>x!>!>},<!><u!>,<u!>>}},{{},{{{}}},{{{<o{!u!!u!>,<!>},<!!!!!>!>,<!>},<i!>,<i!>>},<!>!>u!!a>}}}},{}}},{{{{<!>e}'xoi!!!>ax!>,<ooo!!}o>},{<'}>},{<x!>},<<u,{{'ee'e>,<!!!>{!u!i!e!!uee!i!!axu{>}},{<{{!,!u!!,!!<x!!{>,{}}}}},{{{{<e}!!!>!!!>,'!!,>},<>},{{{}}}}}},{{{<>},{<>,<e!>,<!!!>},<a!!'u!!!!>},{<{>}},{{{<eei!!}!>},<>,<,''e{!!{i!!'>},{{<!!!i'!!a!!!>!e}!>,<u!>,<{>}},{<!!!>},<!>!!,a!!'!!}e'!e!>},<!!i!!'u{u!>,<u>,{<,!!a,aoo!>!!,{!>},<i!!!>}!!,o!'>,<>}}},{{},<}u!!a>},{{{{{<'<a!>,<!>},<!>},<!!!!!!!!'},a!!>}}},{{<!}{!>o!'o!>,<!!!><!!ou!>!>},<!e!>!>,<>}}}}},{},{{{<!!!>o!>u!!!>!>>},<>},{{<,,>},<'{ao!i,!!!>e!uax{!x!!!>>},{<x!>},<a!!!>ex,'!!!>},<'oo!!!>!>>}}},{{{<o''},{!!!>>,{<{!!!!!>}!!!>a!<}!!ea!>!!>,{<!{!!!u!!a<!>,<,!>},<!>,<e!!a>}}}},{{{<oe}'xe'xa,!!!>i!!u,!i!i>},{{<!!!>}>},<a{!a!>},<!>,<!!}!>,<a!>a!>>}},{{},<!>,<>},{{<!!!ae!!!>i'o!!xx>},{<x>}}},{{<!!ia!>},<}!!!>xi!>!!!!u!!!>x!{u'!!e'!!x!!>,<oei!!!><<!!!!i'}i>},{<x!>!>,<,!!>},{}},{{{{}}},{{},{<,!!!>u>}}}}},{{{{},<}!>ia,!>!>},<i{{}!>},<>},{{<>}}},{{<!!,!!!!!>'<!!ee!>},<,>},{}},{{{{{{<>},{<!>,<!!a>}},{{{<,o!!{!>xi!>!<!!}{!!'!!x<!>},<,>}}}}},{{<i}!>},<!>'!!exx>,{<},!!u!>,<!>,<x'!!!>!>>}}},{{<!>!!!>!!!i!o!!{!!e!!xu!iaa!>,<}o,}ao>}}},{{{<!!u{o>,<!}'o!!!>o}a!!}ui!!!!!>!>},<!>>},{<!!ao>,{<o>}}},{{{},{<!>},<!'!>},<,,!x'!>,<>}},{{<x!>},<!ou>,<o!'!,!>!{,!}e!!!u!}!o>},{<!<!!!>,<{{o<!>},<'>}},{{}}},{{}}},{{},{},{<{!!!>,<i!>},<'!!!!o!>},<o{!!!x!!!>}!i>,{}}},{{<}!!!>},<!e'a>,{<!!!!xea{'i!}<<ui!!!!>}},{{<},!>},<!!!!!>,<'!!!!e!>!>,<x<!>},<!>},<>,{<<}<>,{}}},{{},<!u,>}},{{<{<!!!>},<!>},<!!!><!>,<!eo,!>},<u,'!>,>},{<>}}}},{{},{{<ax!!x!>!!!>o!!!><'{'!!!!!!},i'xuu>},{{{},<o>},<'<!!!>!!!i}!>}!!!>!>u!x!!!!}!!!>!x!>'!>,<>}},{{<i!!!!!>!e!!!!a'!!!!}o'}<!>,<e!,>,<!>,<<ooui!!,!!!!}!!<'o!>>},<o<!!<x{!o!!ax!!!!o{<<a!!!!a!!!!!>>}}},{}}},{{{}},{{{},{{<!!!>},<!>},<,!>},<!!!>{!!{i!>},<!>},<!''<!>,<>},<!>},<}!!!>u!ix<!xe!>},<>},{{<e''<o!!!>,}{!>},<e>},{{<!!!>>}}}}},{{{<!!o}uoei>},{{<,oixa>}}},{{<!!a<i!>{x{e!>,<!>},<}<,e>}},{<!>,<i!!oio!>,<x!,o!!!!'!!e',>}},{{<auiex!!!>xu{>,{<}!},}!>,<a!!!!!!e{!>!!,,e}<,{!>>,{}}}}},{{{{<x!>},<xa!>},<!>!!!!a}x<}!!!>!!!!!!x!!!!!>,>},<u>},{{<!!!>,<!{e}x!!u!!!x!!!!>}}},{{{<>},{<!!!>iu!!!!{!>},<>}},{{{<!>},<i>,{{<aa}i>}}},{<!!,o!!!>{!!!>,>,<<u!>,<<!!i<i!!!>ui,!!!!!>,<!>},<}!!!>>}},{{},{{{{},{<ii!>e!>e>}},{<!!{<{a!'{!>!o!>},<!!u}!!<{!{>,{<u!!{o>}},{{{{<!>},<!!!>i!!o!>},<!a}i<,a!>!>},<a!'!!!!!>'>}},{}}}},{{<!>},<!>},<!>},<!!x{<x!}i>},{{}}}}},{{{{},{<'ua<!!!!o!!!!}}}!>,<!!xu>}},{{<!!!x!!!!!!!>},<!!}!!i!>},<!a{!<a'i!!!e',!>,<>},{<{e!!!>!!!>},<{{>}},{{<}e!!}!!!>},<>,{<!!,>,{<>}}}}},{{{},<ao!!u{,oo'!!!!!>!!!>{'!!!>ia!>,<!u,!>},<>},{<!>,<ix!!!>i>}},{{<!!ax!!<,!'{!<xo,>},<!!i!,}!>},<!!!!!!!!{>}},{{{<'uxo!!!>a!!!>,<!!e{{{!>},<,!}!,{>,<!!!!}xxo!!!>!>},<!!!>!!!>a!>,<u!>},<ox!!}ua!!>}},{},{<!>},<<{}!>{!ox!>,<!!!!!>!>,<!>},<>}}}},{{{}},{{{<>,{{<}'i!!o}!!{!!!>}!!}<ea!>!>},<{>}}}},{{<u''{!>,<i!>!,!>},<{'o<!!!><!>!!!,>,{}}}}},{{{<!!x>}},{{<ex>,<},!!!!!>uo}}i!>,<!!!}o!>!!o!>,<,!!x>},{<!>,<e<x{>}},{{<a!><!!!!i!>,<'u!!!>!,u{xx!!{}!!!>,<>,{{{{},{<'oea!e!!,oe!>},<o!!!!!'>}},<u!!u!!i}!>,<!!xa'eu!>,<>},{}}},{<i!i{o!!<ueo{au!>!>},<!}o>}}}}},{{{},{{{{<!>},<,!!!>},<xo!!!>a!!!>!!,!!!!!>e>,{<!>,<a{aa!!!!!>>}},{{{<>}},{{{},<!!!>{!!xx!>,<o!!,'e!>o!!!>!!i!>,<>}}},{{{<}ua!>},<!!!>},<!!!>ee}a>}},{<u,x,u'!>},<!>,<>}}},{{{}}},{{{<i!!!!!>ex>}},{{{{{}}},{{<i!!!>!{!!!>>}},{{{{<a{!!!>!!!!!!e!>,<x!!!>},<!>,<i>},<o}'!>,<!>,<e>},{{<!!!>},u<u!!<!>,<i<,{',!ei!!,>},{<ea'!>},<!!!!!!xe}{!!a,aee,!>,<>}}},{<!!!}'!}u<!>!!'u,aao!>{!>,<>,<'x'!{!>!!u!!!>iax!!!>u!>},<oo>},{{{}}}}},{{<!>,<{!!!>},<!>},<x}a!!!>!x>},<{!!}u}!!!>>}},{{<i}!!!'e!'!>},<a,{ao'}!!!>!>},<!>},<!>},<>},{<!!!>>}}}},{{{{},<!!!>,<!!!>{},!>},<!!<'<e}!!}!>,<!!!>u}{>},{{},{<>},{}},{{},{{},{}}}}},{{{},{{{},{{{<!>x!>},!><u!<!>},<!i>},{<!!,u!>!>!i!>},<a>}},{{<'!>,<!>},<!i!!!!!>x!>xou>},<!>!!!!!!!!ae!>,<'!>},<!!'o!>,<,!>},<{!>,<!>,!>},<>},{{<!!!>!!{<!}!>,<!>ix!>!>,<oo>}}}},{{{<>},<'!>a>},{<!!u!><!!i!!!>>,{{{<!!!!!>!>,<!><{oe<e>},{<a!!!>ea{!>!>,<>}}}},{{<!>,o'!!ix<x!>,!>,<i!!>}}},{{{<!>!!>}}}},{{{<o!!!>!>,<a!!o!!!!i!!!!!>!!eu>}}},{{<!>},<!>!>},<}o}>},{{<}o,!e!!!>,<!>,<a!!<!!''a{!!''o>,{<a!!o}}}u!!!>>}},<xxu!>,<>},{}}},{{{{{{<<>},{<!!!!u'!>!!e}'x!>!!!>i{!>},<>}},{}},<<u!!'i!!!>>}},{{{},{{<<!>!i<!!ei<e!>},<a'>},{<}}!>!!!>!{!<<{{e!!!>!!,ex!!!!{,>}}},<{!!!><'>}},{{{{<!!ui!>o<a>},<i!>},<!!',a!>},<{!>,<!!ui!!ua!>},<>},{{<!>},<o{>,{<!>,<,!>>}},{<!'!!!>!>},<e!>,<{<'!'<!!i!><a!'!!}!!{!>},<>}},{{},{{{<!>},<i!>},<!>,<}!,!>!!<!>uu!!!!u!>},<!e!,u>},<e!aaooi!!!!!>o}!!!>!>,'i}>},<<}a>},{{<>},{}}}}},{{<}}<!!'{xu,!>io<i!>,<>},{<!>'>}}},{{},{{{},{<a!>,<{'!!!!!>e<!!!!oe}!>},<>}},{{{<!!!><o{!!{!>xu!>},<}u!!e>}}},{{{{{},<!!!!xa'!u<!!<!!!>o!>o!!u{{>},{}},{<u,i!!!>,ai!o!>,<uii!>,<!!>,{<,u!!a!!!!uu<e!>},<<i>}},{<,!!{!!i<!!o>,<!>a!!''ue!!x!>i,!}!>,<,a!<>}}}},{{<!!!!!!'!!!>!!xu!>,<e'!>,<>,{}},{{<,,ux>},{{<!!e!>},<e!>},<}}o!>},<<>,{{{<!>,<!>,<x!i!!!>a!!xo<{>,{<!!!>ueo!x>}},{<eio!>},<!!!>!!!>>,{}}},{<!aa!!!>a!'}a!>e!!,!>,<>}}},<!!!<a!>,<o>}},{<}a!>,<!!!i!>},<!!!>x<<>,{}}},{{<{{ax!}{>},{{},{<!!!!!>!>!!!>i!>,<!>ea,>,{<!>,<xi!>ax!!<,,ii'>}}},{{<a!!!>xe!!<!>},<i>,{{{},{{<'!>},<!!!>!!<,<!!eao,,'!>},<!>},<!>},<ax>}}}}},{<!!!>!>},<>}}}}},{{{{<i{'!ie!>},<!>},<a{!!>}},{<!>,<<x!!}!!!{!!<<{!'e!!!<a!!!i!!!>!!!!!>>,{{<,!>},<a{!>ia!!xo!!!!!>}>}}}},{{<x!>,<a!!!u!!!>>,<!!!>!>,<{!{!!i!!!>,<!!'a!>,<xi},!>!>,!!o>},{<!>},<{!!i!>,<,,!!!>,<!!!!'iao!!!>>},{<!>!>!a{'!!!>>,<a!>,<auu<>}},{{{{{},<!>!e!!!!ia}{!>},<{o!>,<!>,<{<xe>},<>},{}},{{<ix<o<!!x!!!>},<!!!>,{!!e<!!!!}>},{<ai!>},<i!>},<u!e!>,<!>},<!>,<!>>}}},{{{{<'>}}},{<!!!>},<{e!!<ou!!!>!><o!>,<>,{{{},<x!!!>!>},<xa>},<!>}e}}!>!>o!>},<a{e{}>}},{{<>,{<,o{!x!!!>eu!>},<xa>}}}}},{{{{{<!!!!i!>},<o}o!>},<<!!!>u!!!>!'!!u'!>,<<!>},<ux>,{}},<}!!!!!>!!!><!!!x>},{<!o!>},<,!>},<}!!ia,!!!a!!x!!'!!!>ea>,{{{<!>,<xeax!!!>io!!!!!><!!!>},<>}}}},{<!'!!<x!!!>},<o!!!>i,<e}}ei!!}!!!>,<}>,<<}o!>},<!>},<>}},{},{{},{{<!!!>!>},<eo!!{!>,!>,<e>,<!!}!>},<a!>,<!!!!!,!!<!!u>}},{{<{!>},<o{<{'>}}},{{{<oox!>,<>}},{<<!!uou<>,{}}}},{{},{{},{{<u{a!!'!>,<xa},'<xo>}}}}}},{{{{{<>},{<}!!a,}<!!<x}u!!!>!!!>!>,<x!!!>u!!!>>,<},!!!>oe!>},<{u!>,<'>}},{<!!exe{!!{<!!!>},<!e!>,<!>,<!>,<ex!!!!>}},{{{{{<e!>},<!!!!!>,<!>},<o,}!{{'>,{{<!!uua!>>,<<iou,!>,<!!!>,o!>,<eo{!e<'!!!>>}}},{<!!!!!>,<!>,<>}},{{{<!!<ou!>},<!!!>,>},<}!!i!!!>!!!>},<!!iii!>>},{{{<a}eo''!ox<!>},<>},{<iouii!>},<u!>},<}!!!><u'!!u!!!>,<u<>}},{{<!>,<!>,<>},<},!!!>ee!>},<!>},<!!>},{}}}},{{{<<''o!>,<}>,{{<!>!>},<!>>}}},{{{<{}!>},<!!a!>{a!!}<<'>},{}}},{{}}},{{{{},{{<!!!>{!!,},{>,{<!!'!!'!>,<ai!e!>,<!!!>>}}},{<!!!>o>}},{{<i!!a}u<!!eou!uo}!>,<!!x!>,<i>}},{<a'!>},<!>,<!a!!!>!!!>u!!!>x!!u>,{}}},{{<!!o!>},<a,}>,{<,!i!!!!!!uoi>}},{<>,{<}{,u!!<!!!!xe'!i!oxu>}}},{<>,{<!>,<ee}{{!!!>!>!!>}}}},{{}}}},{{<,o>,{<>}}},{{{{{{},{{{<!!xi!!!>,<e}u>}},{<!>,<>}}},<<!>},<!!}!>},<u!'!>,<!,u>},{{},<x!!!>,<i}!}!i!'!!!x!!<!>!!!!}<!!>}},{},{<!!!>!>>,<!!!>{aa!!!!a!}e,x!>,<}<<!!!a!>},<e<>}},{{},{}}}}}},{{},{{{{{{{<u,x<e!!>}}},{<>}},<!>,<e!>,<!>},<!>},<axeaea!!oa}!!<>},{{<'!>,<!>},<!!ou!>},<<>}},{{<!!!>u!}!>},<au!x!>!}>},{<i,<o!!!>u>}}}}},{{{{},{<}uxi!!!>!!a!!u!!a{u'!!e<>},{{<>}}}},{{{{{}}},{<>,<!>{!i,}ie!!u!>,<o}!>,<!>,<>},{{{},{}},{<e'!!uxioi!!e!!o!!>}}},{{<x}!!x!!!>xxx!>,<>,{<x!!!>},<!'!!o!>},<{'a}!!a!!!>eo{!!!!!>!>,<,>}},{<a!!xa{>,{}},{<!>,<i<x!!u}<'iu>,<!>!>eo!!!>x,<!!o!!e>}},{{{{{<>}}},{<!!>,<!!,!!!>},<oo!>,<!,!!!>a!>>}},{<!>,<!>>}},{{{{},{<!!e!>!>!>!>},<u!>,<<!>,<!>,<!!!!!>!u>}},{{{{<,o!>a>,<!!!>,}!!i>},{<{>,<x',e!!!>},<ie!!!>{!>},<!u!>,<}u!>},<a'>},{{},{<>,{}}}},{{<x!>,<e!u!!o!>u!>},<>},{{{<{!>!!xi!>,<<x!>i!!xa}}<!!!>xi>}},{{<!!}<!>},<i!>,<u!>},<uo!!!!!!!>'<>,{{}}},{{}}},{<u!>},<}a>,{{<o<eiu!>},<!>xu>}}}},{<o<o<<a!!u>,<<!>,<>}},{{{{<!x!!!>o!a<}!>x>},{}},{{},{{<!!ox!>!!!!{,{x!!}'!!<!!!>,<>}}}}},{{{{},<!,!>!!!!i!!a}o!!u!!'!>,<{!,ouo,e>},{<!>},<!>,<!ua!!!!!>,<!!'e!>}!>!!!><>,{{<!{!e!!'!!!>o!!!>{<!!!>},<>},{<}{o!<!!!>,a!>!!!>>}}},{{<!'ae{!>},<>},<xoux!>},<!!'{!!!>>}},{{<,>}},{<i!!!>!!!!!>},<!!o!!}<>}}},{{{<}o!!!>},<,}a!}e!!!>'>},{}},{<{x{!!{!!,>},{{{<!!e,!!,a!>,<>,{}},{<!>},<a{xie'<ia!>>,{<!!!>,<xe{!>,<!>,<e!<i!>,<<!>,<!!}>}},{<oa!!!!'!!!>e{<!!!e'<!!<!>},<!!u!>},<oax'o>,{}}},{}}},{{{<!>,<!!x}oei!!'!>{<'{!!}e<!>},<!}>,<!!a!<<xu!!a!>,<>},{{},{<!!}a>}},{{},{<a!!!>'!>},<!!!>!!!>u<{!u>},{{{}},{{},{{},{<<!>!>>}}}}}},{{{{<xu!>},<!<!>>,{}}},{{<x!>x,,>,{<!!!>'u!>'<{u!>u>}},{{<e!!i'i!!o,!!!>,<i!>,>}},{{{<!!!>!>,<!>},<u}'!!,{!!<!!!>!>,<!>>},{<{,x!!x'i>}}}},{{{{{},{<!ee!>i,<!>},<,>}},{}},{{<!!!!,!>>},{{<!!!u,!>,<!}ao{>},<!,!>o<e!>!>},<!'!>},<!!!>!>},<i{{!'>}},{<<xoie!>},<!>!!u<oo,!!,o>,{}}},{<!!''!>'!!!>xo,u}>},{{{{}},{{<!>},<oi!>},<<!!!>!!!<x!>,<}!!!>>},{<!!!!a{>,{<{e!!!>'!!!>'!!e{ue'!!o!!!!!>!!!!!>!>,<<>}}}},{<{!!!>u!!!!u!!!!'>,<}{i!!,ox>},{<xeaxx!>e}!>,<<a,!!!!!>},<x>}}},{{{{<>},{}}}}},{{{<u!!!>!!,!>},<''ue,>}},{<!>'{o!>!!!!!>!!x>,<<}o!o{!>},<ue>},{{<u!>}<!>,<!!!!!>>},<a!>},<{'o!!,,x'<!e!!!>>}}},{}},{{{<a{!!!>!>},<{!!>}},{<!!,!<ax!>!!!>},<!!,!!>,{}},{{{<!>},<e!!!!!>uu>}},{{{<,!!!>!!}x!>},<{'!>!>a}!!a!!'a!>{>}},{{{{}},{{}}},<<!!ooa!'a!!o{>},{{<'>},<i,<!}'{x!!!>e<!>},<e!!!>!!<<>}},{{<,e{!!!>a!!<!!,!>},<x!!}iu!!}!>i<>}}}}}},{{{},{{{},{{{<oie>}}}}},{{<!!e!>x!>i!!>,<e!!!>}io!!!>!>,<,<!>},<!>u!!!>>}}},{{{{<a<}'!>o'!>i!!a!,!!>}},<e!'>}},{{{{{{<,!>,<!x!!<!!uo!i}!!''!>,<,'>}}}},{{<'u'!!'x!aou{e}!!!>,<<>},<{a!>,<e!}<}!>},<!>'!o}>},{}},{{{<ao!!!>e>},{{<>},{}}},{{{}},<,!>,<>},{{{<x'{i,'!!!>,<x!>,<u!!!i!>,<!ioo!!!!!!!>!i>}},<a{,}!!!>!!<u{!!!>aui!!}>}}}},{{<!!!>x!!{<>},{{<>},<x!>},<{}{e<o}e!!xxau,'!>},<!!!>>},{{<!>},<},>},{<u!}a!>!>!!xiu!!axu}>}}},{{{{<!>,<!>},<'a!>,<e!>>},{<!!<>,<!>,<!!!>,<oi'{,>}},{<{x!<!x!>},<!a'{!!u!>},<,u,!>,<!>},<!!}}>}}}}},{{{{<u<i!>},<!!!>}x!>!!ixe!!!!>}}}},{{{{<o'!>!!<!!<>},{<a!><!>,<!!<e<!!>}}},{{{{<'{i>}},{<e{{!i!!iea>}}},{{}}},{{{}},{{{{{{},{<!{'!>},<}!!!!e{!}<!>,<u}o!>i>}},<}{>}},{{<',!>},<!x!>},<!!,<!,ee!>e!e}>}}},{{<'e!!'!>}}>,<!!'!>,<i!!e!!!>!}',!!u{iei!<!!>}},{{}},{{{{<!!e,{u!<}!!!'}a>},<{!>},<''i}{!u!!!>!>,<a>}}}}}},{{{{{{{<x!!!!}uxo!>},<u!!u!!!>!>},<!!}!>},<!!<e>}},{<e{!!!!o,!>!!!>},<!<!>,<<!>},<{!>},<>}}},{{<'e!!a'!>},<i}>}},{<u<i,>,<,o,}{!!x>}},{},{{{{{<!!!>>,{<!!,}e>}},{<!>},<!>!>},<},!!!!}i!!,!>},<!>ao!!e'<}>}},{{{<}>},<!!!!a!!!>ux>}}},{{<!>},<<ia>,<!>},<o!>!!!>},<oe!>},<a,i>},{}},{{{{},{{{<,'{'}!!!!e!!{>},{<,!a'a!,!!!>>}},<e{!!,x,e!!!!<i!>!!!>,<!!<!>}!!}'>},{<!!i!>,<,!e<!,a>,{<>}}},{{}},{{<>}}},{{<a>},{}},{{{{<xi{}'!>},<{<i!>,<!>x{!!!>!!oe{{o>}},{{},{<ae}!>!>!!!>!<}x!>>}}}},{{{},{<<!>},<{>}},{<a}!}!>,<!!!>},<,{i>,<!a{!>,<!>,<},!!!!!!!>xua!!}>}}}}}},{{},{{{},{{{<i>},{<!!a!!'!!!>x!!!>o{'!!ux!!x!!!!!!',>}},{<i}!{e!>},<!!eu!>,!!!>ea>,{}},{{{<,'{xe,!>},<!o'<,i!!'!!{x!!!>>},{<!>},<}{!>,<a!>},<!>o!'!>!<a!!!>!!!>{uo>}},<iou!!!>,<e},exi!!!>!}'!!{!uxa>}},{{<ee!oeae}!!!>auex!>},<>,<e>},{<o!!o!!!>!!!!!>>},{{<!!!ui,}>}}}},{{{{<!<!!!>,<!>},<{i!!!>i>}},{{<'a}>},<{ix!!a!ax!!!!!>'!>,<<!!!!!>!>},<{>},{}},{{{<!!!!!>x!>,<'!!}exaxa!>,<!<u'!>},<x}>,{<e!>},<!>!!xx,!!<!!}e!!u>}},{{{<!>!!!>!>},<,!,!!<ix{x}<x}!!!>!'!a!!<>}}},{{<aae!!!>,<'!<>},{}}},{{{{{<!>},<!!!>>}},{}}},{{<!u!!,!!!>ee<!!!!<!!{!>},<io!>,<>},{<}}iu!>},<e!!a,}<!>i!!!>!>!!>}}},{{{{}},{<a<e<!!!!!>,<!}!>},<!>},<u!!u!>{}'>}}},{{<u!!!ee'{xe!>!>,<e!!'!!!>!>,<a!!,x,,!!,>,{<u''xo!>!>!>oi{!!>}},{{<<i{>},{}},{{{}}}}},{{{},{{<'x!!>},{<i>}},{{<!!!!!>'x!!!>i'<oe!e!!iu'!!!!!!!>>}}}},{{{{<x}!>i!!o!!!>u,!>!i!!!>ii>},<!!,!>,<oai!io!!i!>},<!!!<e<e!!x!!!>,<>},{{<!>},<<!>},<!!i!!a!xo!!e!>u!<!'!!!>,<<>},<e!u!!!>aiu<!>!>!!!!e!!!>,<!!'e!>!!>}},{{<!><!>,<!>},<!>},<,u<,}!>},<!!!!e,o'>,{{},{<a>}}},{}},{{{{<x,<!!xo!>>},{<ox!!<{}',!><>}},{<!!!>'i!!!>,<i>}}}}},{{{}},{}},{{{<a}ei{o!>,<!!!!ui!'!<{!>,<!>},<!>i>},{<<!!!>,<}}ao!>},<!!!>},<!!!>>}},{{{<!{u!!!!!>!}!!'x!>,<<x!>,<!!!>'>},{}},{{{<!,!eu!>'u!!{!!!>e}'o!e!!!!!>o>},{<x!!!>}!<i!!!>!'!!!uooeo!a!x!iio!!>}},<o!!!{ex}!!!>!u!x{,!>!>!>},<u!'>},{<!>},<a!a}!!!>{!i!!!!auo<i>}},{<x'!>},<{e!!!>!!uu!>,<!!a>,{<oxo!>},<!>,<!x}xo!{'{,!>},<>}}}},{{{<!!{xxx!!,>},<!i<<>},{{<!!ox!e>}},{{{<e}!!!}>}},{<<!!!>!>,<!>,<!u!!!!!!,u!!!!!!!!!!!>,<'!>x!!!>{>}}}},{}}}";
    }
}